src/env.h

```diff
@@ -92,13 +92,17 @@ struct PackageConfig {
   enum class Exists { Yes, No };
   enum class IsValid { Yes, No };
   enum class HasMain { Yes, No };
+  enum class HasExports { Yes, No };
   enum PackageType : uint32_t { None = 0, CommonJS, Module };
 
   const Exists exists;
   const IsValid is_valid;
   const HasMain has_main;
   const std::string main;
   const PackageType type;
+
+  const HasExports has_exports;
+  const v8::CopyablePersistentTraits<v8::Value>::CopyablePersistent exports;
```

Comment:
Can you keep using `v8::Global` here, for consistency with the rest of core? There doesn’t seem to be a need for copying semantics here (you do need to use `std::move(exports)` instead of `exports` when initializing the `PackageConfig` below in that case, but that seems to be correct anyway).

Reply:
@guybedford Do you remember why you went with this implementation over `v8::Global`?

Reply:
I think because I didn't know how to use `std::move` so this was my way of working around that initialization failure!?

Reply:
Thanks for the context! I’ll try to make it work with `v8::Global`. 🤞

Reply:
After switching to `v8::Global`, the next issue seems to be that `package_json_cache.emplace` doesn’t like the data structure with a `v8::Global` field. The expansion is pretty long but starts with:

```
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/memory:1805:31: error: no
      matching constructor for initialization of 'std::__1::pair<const std::__1::basic_string<char>, const
      node::loader::PackageConfig>'
            ::new((void*)__p) _Up(_VSTD::forward<_Args>(__args)...);
                              ^   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Not sure yet how that could be resolved.

Reply:
I believe that was exactly the original issue I had.

Reply:
@jkrems Not really sure what to  do with a partial error message here … :smile:

Reply:
@addaleax That’s more than fair. :D I pushed my changes to this PR and also created a gist with the output I get locally: https://gist.github.com/jkrems/7ff6e9b755288f806eefb11010935bc3. I may continue to dig into this later today.

Reply:
Current interpretation:

1. The move constructor provided by `v8::Global<T>` isn’t compatible with the default member initialization generated for structs and/or the struct constructor gets confused by the reference/copy constructor. The end result is that the move constructor of `PackageConfig` gets marked as deleted.
2. I can define a custom move constructor but then I’ll have to also provide a general constructor since it removes the aggregate initialization / brace constructor.

*phew* I’m never gonna underestimate the changes in C++11 again. I’ll go forward with switching this code to an explicit field constructor + move constructor.

Reply:
Okay, turns out that brace-syntax is smart enough to use the constructor with each field spelled out. Neat! I *think* this is all working now. Not 100% sure if implementing them in `env-inl.h` is the appropriate place but that’s where similar code seems to have ended up.